src/effects/VST3/internal/ComponentHandler.h

```diff
@@ -0,0 +1,73 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file ComponentHandler.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <atomic>
+#include <functional>
+#include <pluginterfaces/vst/ivsteditcontroller.h>
+
+namespace Steinberg
+{
+   namespace Vst
+   {
+      class IParameterChanges;
+   }
+}
+
+namespace internal
+{
+   /** Stores all pending changes from the editor, and
+     * safely transfers them to the processing thread on request.
+     * It's lock and wait free (given that there are two threads),
+     * but it's theoretically possible that
+     * processing thread will never get pending changes, but in real life
+     * should never be the case.
+     * Current implementation isn't strict about begin/perform/endEdit
+     * order correctness(actually it's completely ignored right now,
+     * but that might change in future).
```

Comment:
Is that last sentence still true?  I think all of that has been fixed now.

Reply:
This comment relates to the `ComponentHandler` class itself, not the client code which uses it.